project/source/Scripts/Objects/SourceContainer.gd

```diff
@@ -10,10 +10,10 @@ enum ContainerType {ERLENMEYER_FLASK, MICRO_CENTRIFUGE_TUBE} #TODO: I'm not conv
 @export var image: Texture2D = null: set = SetOverrideImage
 
 @export var substance: PackedScene = null
-@export var substance_parameters: Array = null
-var contents = null
+@export var substance_parameters: Array
```

Comment:
The type of the elements in this array should be specified. From what I can tell, the only substance that supports parameters is `DNASubstance`, which keeps an `Array[float]` of parameters. We should probably find a better way to handle this so that `SourceContainer` doesn't have to know about this.